staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -287,6 +289,7 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 		}
 	}()
 
+	retryInternalError := NewRetryWithDeadline(r.MaxInternalErrorRetryDuration, time.Minute, r.clock)
```

Comment:
I was thinking a bit more about this and I think I can leave with explicit retries of all internal errors (if explicitly opted-in).

But I wanted to propose to change the interface of this retry to be more consistent with what we decided in other places, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/with_retry.go#L54

We can probably have something visibly simpler here like:
```
type RetryWithDeadline interface {
 After(error)
 ShouldRetry() bool
}
```

And then I would expect it to be called like:

```
err := watchHandler(....)
retry.After(err)
if err != nil {
  // the logic goes here
}
```

Also - can you please move the RetryWithDeadline implementation to a separate file?

Reply:
Done.